File: apps/react-native/src/repositories/MessagesRepository.js
/**
 * Messages Repository
 * Equivalent to Flutter's MessagesRepository
 * 
 * Manages caregiver messages
 */

import { CaregiverMessage } from '../models/CaregiverMessage';

class MessagesRepository {
  constructor() {
    this._messages = [
      new CaregiverMessage(
        'm01',
        'Care Team',
        'New care plan note',
        'Updated instructions for evening medication.',
        new Date(2026, 0, 31, 21, 45),
        true
      ),
      new CaregiverMessage(
        'm02',
        'Dr. Patel',
        'Visit follow-up',
        'Please monitor blood pressure twice daily.',
        new Date(2026, 0, 31, 21, 15),
        true
      ),
      new CaregiverMessage(
        'm03',
        'Clinic Scheduler',
        'Appointment confirmed',
        'Appointment confirmed for 2026-02-01 0900.',
        new Date(2026, 0, 31, 20, 30),
        false
      ),
      new CaregiverMessage(
        'm04',
        'CareConnect Alerts',
        'Medication missed',
        'Morning insulin marked as missed.',
        new Date(2026, 0, 31, 19, 45),
        true
      ),
      new CaregiverMessage(
        'm05',
        'Care Team',
        'Lab results available',
        'New lab results have been uploaded to the chart.',
        new Date(2026, 0, 31, 18, 0),
        false
      ),
      new CaregiverMessage(
        'm06',
        'Dr. Nguyen',
        'Blood sugar logs required',
        'Please enter blood sugar readings for past 3 days.',
        new Date(2026, 0, 31, 17, 30),
        true
      ),
      new CaregiverMessage(
        'm07',
        'Pharmacy',
        'Prescription ready',
        'Your refill for Metformin is ready for pickup.',
        new Date(2026, 1, 1, 9, 10),
        false
      ),
      new CaregiverMessage(
        'm08',
        'CareConnect Alerts',
        'New symptom entry',
        'A new symptom was logged: dizziness.',
        new Date(2026, 1, 1, 10, 5),
        true
      ),
      new CaregiverMessage(
        'm09',
        'Dr. Rivera',
        'Medication adjustment',
        'Reduce evening dose by half starting tonight.',
        new Date(2026, 1, 1, 11, 25),
        false
      ),
      new CaregiverMessage(
        'm10',
        'Lab Services',
        'Fasting reminder',
        'Reminder: fasting is required before tomorrow’s blood draw.',
        new Date(2026, 1, 1, 12, 0),
        true
      ),
      new CaregiverMessage(
        'm11',
        'Clinic Scheduler',
        'Appointment moved',
        'Your appointment has been moved from 2:00 PM to 3:30 PM.',
        new Date(2026, 1, 1, 13, 40),
        false
      ),
      new CaregiverMessage(
        'm12',
        'Care Team',
        'Care plan review',
        'Please review updated meal and hydration recommendations.',
        new Date(2026, 1, 1, 14, 15),
        true
      ),
      new CaregiverMessage(
        'm13',
        'Home Health RN',
        'Visit notes',
        'Patient tolerated mobility exercises well. Mild fatigue reported.',
        new Date(2026, 1, 1, 15, 55),
        false
      ),
      new CaregiverMessage(
        'm14',
        'CareConnect Alerts',
        'Missed check-in',
        'Daily wellness check-in was not completed.',
        new Date(2026, 1, 1, 16, 20),
        true
      ),
      new CaregiverMessage(
        'm15',
        'Dr. Hassan',
        'Follow-up question',
        'Has swelling improved since last week?',
        new Date(2026, 1, 1, 17, 45),
        false
      ),
      new CaregiverMessage(
        'm16',
        'CareConnect Support',
        'Ticket update',
        'Your support request has been received and is being reviewed.',
        new Date(2026, 1, 1, 18, 30),
        true
      ),
      new CaregiverMessage(
        'm17',
        'Nutrition Services',
        'Dietary guidance',
        'Please limit sodium intake to help manage blood pressure.',
        new Date(2026, 1, 2, 8, 10),
        false
      ),
      new CaregiverMessage(
        'm18',
        'CareConnect Alerts',
        'New task assigned',
        'A new caregiver task has been assigned: medication audit.',
        new Date(2026, 1, 2, 9, 0),
        true
      ),
      new CaregiverMessage(
        'm19',
        'Clinic Scheduler',
        'Transportation note',
        'Transportation confirmed for the 11:00 AM visit.',
        new Date(2026, 1, 2, 9, 35),
        true
      ),
      new CaregiverMessage(
        'm20',
        'Dr. Williams',
        'Lab review',
        'Lab results reviewed. No urgent concerns at this time.',
        new Date(2026, 1, 2, 10, 50),
        false
      ),
      new CaregiverMessage(
        'm21',
        'Care Team',
        'Pain level update requested',
        'Please update today’s pain score in the chart.',
        new Date(2026, 1, 2, 11, 15),
        true
      ),
      new CaregiverMessage(
        'm22',
        'CareConnect Alerts',
        'Device disconnected',
        'Blood pressure device has not synced in 24 hours.',
        new Date(2026, 1, 2, 12, 30),
        true
      ),
      new CaregiverMessage(
        'm23',
        'Home Health PT',
        'Exercise plan',
        'Please continue balance exercises twice daily.',
        new Date(2026, 1, 2, 13, 5),
        false
      ),
      new CaregiverMessage(
        'm24',
        'Care Team',
        'Medication refill needed',
        'Reminder: medication supply is low. Please request a refill.',
        new Date(2026, 1, 2, 14, 25),
        true
      ),
      new CaregiverMessage(
        'm25',
        'CareConnect Alerts',
        'Upcoming visit reminder',
        'Upcoming visit scheduled within the next 24 hours.',
        new Date(2026, 1, 2, 15, 40),
        false
      ),
      new CaregiverMessage(
        'm26',
        'Billing Office',
        'Statement available',
        'A new billing statement is now available in your account.',
        new Date(2026, 1, 2, 16, 15),
        true
      ),
    ];
  }

  allMessages() {
    return [...this._messages];
  }

  unreadCount() {
    return this._messages.filter(m => m.unread).length;
  }

  unreadMessages() {
    return this._messages.filter(m => m.unread);
  }
}

export const messagesRepository = new MessagesRepository();

File: apps/flutter/lib/core/messages/messages_repository.dart
// lib/core/messages/messages_repository.dart

import 'caregiver_message.dart';

class MessagesRepository {
  MessagesRepository._();
  static final MessagesRepository instance = MessagesRepository._();

  final List<CaregiverMessage> _messages = [
    CaregiverMessage(
      id: 'm01',
      patientId: 'p01',
      sender: 'Care Team',
      subject: 'New care plan note',
      preview: 'Updated instructions for evening medication.',
      sentAt: DateTime(2026, 1, 31, 21, 45),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm02',
      patientId: 'p02',
      sender: 'Dr. Patel',
      subject: 'Visit follow-up',
      preview: 'Please monitor blood pressure twice daily.',
      sentAt: DateTime(2026, 1, 31, 21, 15),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm03',
      patientId: 'p01',
      sender: 'Clinic Scheduler',
      subject: 'Appointment confirmed',
      preview: 'Appointment confirmed for 2026-02-01 0900.',
      sentAt: DateTime(2026, 1, 31, 20, 30),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm04',
      patientId: 'p03',
      sender: 'CareConnect Alerts',
      subject: 'Medication missed',
      preview: 'Morning insulin marked as missed.',
      sentAt: DateTime(2026, 1, 31, 19, 45),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm05',
      patientId: 'p04',
      sender: 'Care Team',
      subject: 'Lab results available',
      preview: 'New lab results have been uploaded to the chart.',
      sentAt: DateTime(2026, 1, 31, 18, 00),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm06',
      patientId: 'p05',
      sender: 'Dr. Nguyen',
      subject: 'Symptom check-in',
      preview: 'How is the chest pain today? Any changes?',
      sentAt: DateTime(2026, 1, 31, 17, 30),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm07',
      patientId: 'p06',
      sender: 'Home Health',
      subject: 'Visit notes',
      preview: 'Summary: mobility improving; continue exercises.',
      sentAt: DateTime(2026, 1, 31, 16, 15),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm08',
      patientId: 'p02',
      sender: 'CareConnect Alerts',
      subject: 'New message from provider',
      preview: 'You have a new message regarding upcoming visits.',
      sentAt: DateTime(2026, 1, 31, 15, 00),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm09',
      patientId: 'p01',
      sender: 'Clinic Scheduler',
      subject: 'Reschedule request',
      preview: 'Would you like to move Mark Lee to 4:30 PM?',
      sentAt: DateTime(2026, 1, 31, 14, 30),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm10',
      patientId: 'p07',
      sender: 'Care Team',
      subject: 'Diet reminder',
      preview: 'Low sodium meals recommended this week.',
      sentAt: DateTime(2026, 1, 31, 13, 45),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm11',
      patientId: 'p03',
      sender: 'CareConnect Alerts',
      subject: 'Blood pressure alert',
      preview: 'Elevated reading recorded. Review recommended.',
      sentAt: DateTime(2026, 1, 31, 12, 15),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm12',
      patientId: 'p02',
      sender: 'Dr. Patel',
      subject: 'Medication adjustment',
      preview: 'Adjust dosage starting tomorrow morning.',
      sentAt: DateTime(2026, 1, 31, 11, 30),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm13',
      patientId: 'p08',
      sender: 'Home Health',
      subject: 'Supplies delivered',
      preview: 'Supplies delivered to the front desk.',
      sentAt: DateTime(2026, 1, 31, 10, 00),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm14',
      patientId: 'p09',
      sender: 'Care Team',
      subject: 'Daily summary',
      preview: '3 patients need attention. 12 visits scheduled.',
      sentAt: DateTime(2026, 1, 31, 9, 15),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm15',
      patientId: 'p01',
      sender: 'CareConnect Alerts',
      subject: 'Task overdue',
      preview: 'Caregiver task overdue: confirm appointment details.',
      sentAt: DateTime(2026, 1, 31, 8, 45),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm16',
      patientId: 'p10',
      sender: 'Clinic Scheduler',
      subject: 'Check-in instructions',
      preview: 'Arrive 10 minutes early and bring ID.',
      sentAt: DateTime(2026, 1, 31, 8, 00),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm17',
      patientId: 'p04',
      sender: 'Care Team',
      subject: 'Care plan update',
      preview: 'Updated fall risk precautions added.',
      sentAt: DateTime(2026, 1, 31, 7, 30),
      unread: false,
    ),

    // -----------------------------------------------------------------------
    // NEW (8) — total now 25
    // -----------------------------------------------------------------------
    CaregiverMessage(
      id: 'm18',
      patientId: 'p02',
      sender: 'Care Team',
      subject: 'Caregiver note requested',
      preview: 'Please add an update after the evening routine.',
      sentAt: DateTime(2026, 1, 31, 7, 00),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm19',
      patientId: 'p03',
      sender: 'Clinic Scheduler',
      subject: 'Appointment time change',
      preview: 'Appointment moved to 10:30 AM tomorrow.',
      sentAt: DateTime(2026, 1, 31, 6, 30),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm20',
      patientId: 'p05',
      sender: 'CareConnect Alerts',
      subject: 'New lab order',
      preview: 'A new lab order was placed. Please review instructions.',
      sentAt: DateTime(2026, 1, 31, 6, 00),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm21',
      patientId: 'p07',
      sender: 'Dr. Nguyen',
      subject: 'Medication question',
      preview: 'Did the patient take the morning dose as scheduled?',
      sentAt: DateTime(2026, 1, 31, 5, 30),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm22',
      patientId: 'p01',
      sender: 'Home Health',
      subject: 'Visit confirmation',
      preview: 'Home visit confirmed for tomorrow morning.',
      sentAt: DateTime(2026, 1, 31, 5, 00),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm23',
      patientId: 'p08',
      sender: 'CareConnect Alerts',
      subject: 'Refill reminder',
      preview: 'Medication supply low. Consider requesting a refill.',
      sentAt: DateTime(2026, 1, 31, 4, 30),
      unread: true,
    ),
    CaregiverMessage(
      id: 'm24',
      patientId: 'p09',
      sender: 'Care Team',
      subject: 'Diet follow-up',
      preview: 'Please track fluid intake for the next 48 hours.',
      sentAt: DateTime(2026, 1, 31, 4, 00),
      unread: false,
    ),
    CaregiverMessage(
      id: 'm25',
      patientId: 'p10',
      sender: 'Clinic Scheduler',
      subject: 'Paperwork needed',
      preview: 'Please complete intake paperwork before arrival.',
      sentAt: DateTime(2026, 1, 31, 3, 30),
      unread: true,
    ),
  ];

  List<CaregiverMessage> all() {
    final items = List<CaregiverMessage>.from(_messages);
    items.sort((a, b) => b.sentAt.compareTo(a.sentAt));
    return items;
  }

  int unreadCount() => _messages.where((m) => m.unread).length;

  List<CaregiverMessage> topN(int n) => all().take(n).toList();
}